test/framework/src/main/java/org/opensearch/repositories/blobstore/OpenSearchBlobStoreRepositoryIntegTestCase.java

```diff
@@ -344,9 +345,11 @@ public void testSnapshotAndRestore() throws Exception {
         // higher timeout since we can have quite a few shards and a little more data here
         ensureGreen(TimeValue.timeValueSeconds(120));
 
-        for (int i = 0; i < indexCount; i++) {
-            assertHitCount(client().prepareSearch(indexNames[i]).setSize(0).get(), docCounts[i]);
-        }
+        assertBusy(() -> {
+            for (int i = 0; i < indexCount; i++) {
+                assertHitCount(client().prepareSearch(indexNames[i]).setSize(0).get(), docCounts[i]);
```

Comment:
Do we need assertBusy on `assertHitCount` statement ?

Reply:
yes, as assertHitCount is making an API call, it's response might change in a assert busy.

I have updated it to something like this :
```
for (int i = 0; i < indexCount; i++) {
            final int j = i;
            assertBusy(() -> {
                assertHitCount(client().prepareSearch(indexNames[j]).setSize(0).get(), docCounts[j]);
            });
        }
```